pkg/kubectl/cmd/expose.go

```diff
@@ -182,6 +182,17 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 			params["ports"] = strings.Join(ports, ",")
 		}
 	}
+
+	// Always try to derive protocols from the exposed object, as
+	// user may specify UDP protocol
+	protocolsMap, err := f.ProtocolsForObject(inputObject)
```

Comment:
@kargakis if we follow this pattern, we have to leave the default value of `protocol` to empty, now it is `TCP` and which just makes expose multi protocols not ok.